pkg/resource/notebook_instance/custom_pre_delete.go

```diff
@@ -0,0 +1,30 @@
+package notebook_instance
+
+import (
+	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
+)
+
+/*
+This code stops the NotebookInstance right before its about to be deleted.
+Returns True if the Notebook was stopped
+*/
+func (rm *resourceManager) customPreDelete(
+	r *resource) bool {
+	latestStatus := *r.ko.Status.NotebookInstanceStatus
+	if &latestStatus == nil {
+		return false
+	}
+
+	//We only want to stop the Notebook if its not already stopped/stopping or not in a failed state.
+	if rm.isInServiceStatus(latestStatus) {
+		err := rm.stopNotebookInstance(r)
+		if err == nil {
```

Comment:
1. Error needs to bubble up, can't ignore it

2. I think it'll be cleaner to write code directly in the template instead of customPreDelete. also do you need a method if just checking for InService? previously it was a bunch